test/services/helpers/CreditHistoryDataHelper.scala

```diff
@@ -112,22 +160,38 @@ trait CreditHistoryDataHelper {
     balanceDetails = Some(expectedBalancedDetails)
   )
 
+
+  val creditDetailModelasSetInterest = CreditDetailModel(
+    date = LocalDate.parse("2022-03-29"),
+    documentDetail = documentDetailsWhichIsRepaymentInterestCredit,
+    RepaymentInterest,
+    balanceDetails = Some(expectedBalancedDetails)
+  )
+
   val taxYearFinancialDetails_PlusOneYear = FinancialDetailsModel(
     balanceDetails = BalanceDetails(1.00, 2.00, 3.00, None, None, None, None, None),
     documentDetails = List(
       documentDetailsWhichIsMfaCredit,
-      DocumentDetail(taxYear = 2022, transactionId = "testTransactionId2", documentDescription = None,
+      DocumentDetail(taxYear = 2022, transactionId = "testTransactionId8", documentDescription = None,
         documentText = None, outstandingAmount = None, originalAmount = None, documentDate = LocalDate.of(taxYear, 3, 29)),
       documentDetailsWhichIsBCCredit
     ),
     financialDetails = List(
-      FinancialDetail(taxYear = "2022", mainType = Some("ITSA Overpayment Relief"), transactionId = Some("testTransactionId1"),
+      FinancialDetail(taxYear = "2022", mainType = Some("ITSA Overpayment Relief"), mainTransaction = Some("4004"), transactionId = Some(documentDetailsWhichIsMfaCredit.transactionId),
         transactionDate = None, `type` = None, totalAmount = None, originalAmount = None, outstandingAmount = None,
+//<<<<<<< HEAD
         clearedAmount = None, chargeType = None, accruedInterest = None, items = Some(Seq(SubItem(Some(fixedDate.plusDays(3)))))),
-      FinancialDetail(taxYear = "2022", mainType = None, transactionId = Some("testTransactionId2"),
+      FinancialDetail(taxYear = "2022", mainType = None, mainTransaction = None, transactionId = Some("testTransactionId8"),
         transactionDate = None, `type` = None, totalAmount = None, originalAmount = None, outstandingAmount = None,
         clearedAmount = None, chargeType = None, accruedInterest = None, items = Some(Seq(SubItem(Some(fixedDate.plusDays(5)))))),
-      FinancialDetail(taxYear = "2022", mainType = Some("SA Balancing Charge Credit"), transactionId = Some("testTransactionId3"),
+      FinancialDetail(taxYear = "2022", mainType = Some("SA Balancing Charge Credit"), mainTransaction = Some("4905"), transactionId = Some(documentDetailsWhichIsBCCredit.transactionId),
+//=======
```

Comment:
its seems there are some issues after merging changes from the main, if this merge conflict can be resolved